### ISSUE TITLE
Magento\Email\Model\Transport does not correctly implement Magento\Framework\Mail\TransportInterface

### ISSUE THREAD

**Issue by hatimeria-artur-jewula** [magento/magento2#25353]
### Preconditions (*)
1. Magento 2.3.3

### Steps to reproduce (*)
Consider the following code:
```
/** @var Magento\Framework\Mail\Template\TransportBuilder $transportBuilder */
$transportBuilder
    ->setTemplateIdentifier($template)
    ->setTemplateOptions([
        'area' => Area::AREA_FRONTEND,
        'store' => 0
    ])
    ->setTemplateVars($params)
    ->setFromByScope('general')
    ->addTo($recipient);
    
/** @var Magento\Framework\Mail\TransportInterface $transport */ 
$transport = $transportBuilder->getTransport();

/** @var Magento\Framework\Mail\MessageInterface $message */
$message = $transport->getMessage();
```

### Expected result (*)
1. `$mesage` variable value should implement `Magento\Framework\Mail\MessageInterface` (according to phpdoc of `Magento\Framework\Mail\TransportInterface`) 

### Actual result (*)
1. Returned value is an instance of `Magento\Framework\Mail\EmailMessageInterface`

**Comment by sdzhepa** [comment]
Hello @hatimeria-artur-jewula 

Thank you for contribution and collaboration!

It seems already known issue for `2.3.3`.

It was mentioned in Magento Open Source 2.3.3 Release Notes:

- in [Backward-incompatible Changes](https://devdocs.magento.com/guides/v2.3/release-notes/release-notes-2-3-3-open-source.html#backward-incompatible-changes)
- in [Apply the EmailMessageInterface backward compatibility issue patch to resolve an email interface backward-incompatibility issue](https://devdocs.magento.com/guides/v2.3/release-notes/release-notes-2-3-3-open-source.html#apply-the-emailmessageinterface-backward-compatibility-issue-patch-to-resolve-an-email-interface-backward-incompatibility-issue)
- Link to Patch: [EmailMessageInterface backward compatibility issue patch for Magento 2.3.3](https://magento.com/tech-resources/download#download2333)

Also, it was already fixed in `2.3-develop` branch in commit: https://github.com/magento/magento2/commit/e14296f8fc3fbf22d4de6af35e3c72f8678c5841

Could you please confirm that the patch fixes this bug and we can close this issue?

**Comment by engcom-Charlie** [comment]
Hello @hatimeria-artur-jewula 
We are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

Thanks for your report!
